[next-gen docs] Update next-gen capitalization of terms and also trim down on the getting started guide (#1709)
slackapi/bolt-js: docs/_future/getting_started_future.md

# Getting started <span class="label-beta">BETA</span>

<div class="section-content">
This guide will walk you through getting started with next-generation developing Slack apps. This new platform enables creation of modular, shareable, and remixable building blocks within your apps that make it easier than ever to connect and reuse functionality.

You can learn more about the next-generation platform on Slack's [official introduction page](https://api.slack.com/future).
</div>

> 💡 Currently, the new Slack Platform beta is only available for workspaces on a paid plan.

---
### Limitations {#limitations}

Bolt for JavaScript supports app development using next-gen platform features like functions and workflows, development tooling through the [Slack CLI](https://api.slack.com/future/tools/cli), and all generally available [Slack Platform](https://api.slack.com/start/overview) features.

#### Bolt for JavaScript does not yet support:
- [Deployment](https://api.slack.com/future/deploy) of your next-gen app to secure and managed Slack infrastructure

> - The Datastores API [Datastores](https://api.slack.com/future/datastores) and related Datastores functionality

Reviewer: ```suggestion
- The Datastores API [Datastores](https://api.slack.com/future/datastores) and related datastores functionality
```

Once you've logged into the CLI using `slack login` and verified your login using `slack auth list`, you can proceed with the instructions in this guide.

#### Installing Node {#installing-node}

Make sure your machine has the most recent version of [Node](https://nodejs.org/en/) installed. Depending on what operating system your machine uses, you can either install Node through a package manager (such as [Homebrew](https://brew.sh/) for Mac) or directly from the website.

#### Accept the Beta Terms of Service {#accept-tos}

In order to use the next-generation platform features, you'll need to accept a Terms of Service (TOS) for Slack Platform Beta in your Workspace Settings [here](https://slack.com/admin/settings#hermes_permissionfs).

> 💡 You must be an admin of your workspace to be able to access the Workspace Settings and accept the Terms of Service.

If the TOS has been accepted, you will see this view under the "Slack Platform Beta" section when navigating to the above link:

![Beta Terms of Service](../assets/beta-tos-future.png "Beta Terms of Service")

> Congratulations! You're now ready to start building using the [next-generation platform](/bolt-js/future/getting-started#next-gen). 🎉 You can now proceed with creating a new app by following the instructions below or explore how to [set up an existing app](/bolt-js/future/setup-existing-app) on the next-gen platform.

Reviewer: ```suggestion
Congratulations! You're now ready to start building using the [next-generation platform](/bolt-js/future/getting-started#next-gen). 🎉 You can now proceed with creating a new app by following the instructions below or explore how to [set up an existing app](/bolt-js/future/setup-existing-app).
```
Thoughts on removing the second mention of next-gen if it's mentioned in the sentence above?

---

### Create a new app {#create-app}

> To get started with development, you'll need to create a Slack app. 

Reviewer: ```suggestion
To start, you'll need to create a Slack app. 
```

> Run the following command to create a starter next-generation app:

Reviewer: ```suggestion
Create a starter next-generation app with the CLI:
```

🔔 If you leave the workspace, you won’t be able to manage any apps you’ve deployed to it. Apps you deploy will belong to the workspace even if you leave the workspace
```

At this point, your app is now ready to go in the `my-app` directory! 

The starter template your app is based off of includes a "Hello World" example that demonstrates how to use the three main building blocks of next-generation Slack apps: [built-in](/bolt-js/future/built-in-functions) and [custom](/bolt-js/future/custom-functions) functions, [triggers](/bolt-js/future/triggers), and [workflows](/bolt-js/future/workflows).

**At a quick glance:**

> 1. **Functions** are what define the actions of your app

Reviewer: ```suggestion
1. **Functions** define the actions of your app
```

> 2. **Workflows** are made up of **functions**

Reviewer: ```suggestion
2. **Workflows** group **functions** into ordered steps
```

> 3. **Triggers** are how **workflows** are executed

Reviewer: ```suggestion
3. **Triggers** invoke **workflows**
```

If you'd like to test the "Hello World" example included in your template and see how functions, workflows, and triggers all tie together, read on! 📖

> > 💡 Now that it's been set up, this application can be used as starting point for you to build your own next-generation features. You can explore on your own and dive into the code yourself [here](https://github.com/slack-samples/bolt-js-starter-template/tree/future), or continue your journey by diving into [app manifests](/bolt-js/future/app-manifests) or learning how to add more [functions](/bolt-js/future/built-in-functions), [workflows](/bolt-js/future/workflows), and [triggers]() to your app!

Reviewer: ```suggestion
> 💡 This application can be the starting point for you to build your own app features. Explore on your own and dive into the code [here](https://github.com/slack-samples/bolt-js-starter-template/tree/future), or continue your journey by learning about [app manifests](/bolt-js/future/app-manifests), and how to add more [functions](/bolt-js/future/built-in-functions), [workflows](/bolt-js/future/workflows), and [triggers]() to your app!
```

---
### Set up your trigger {#setup-trigger}

> Your app comes with a pre-built [workflow](https://github.com/slack-samples/bolt-js-starter-template/blob/future/manifest/workflows/sample-workflow.js) that uses a pre-written [function](https://github.com/slack-samples/bolt-js-starter-template/blob/future/manifest/functions/sample-function.js). This workflow is executed by a trigger defined in [this trigger configuration file](https://github.com/slack-samples/bolt-js-starter-template/blob/future/triggers/hello-world-trigger.json). We can run a command via the CLI to initialize this trigger so that we can run the workflow.

Reviewer: Can we break this down, possibly by breaking off the last sentence "We can run.." into it's own line.

> Your app comes with a pre-built [workflow](https://github.com/slack-samples/bolt-js-starter-template/blob/future/manifest/workflows/sample-workflow.js) that uses a pre-written [function](https://github.com/slack-samples/bolt-js-starter-template/blob/future/manifest/functions/sample-function.js). This workflow is executed by a trigger defined in [this trigger configuration file](https://github.com/slack-samples/bolt-js-starter-template/blob/future/triggers/hello-world-trigger.json). We can run a command via the CLI to initialize this trigger so that we can run the workflow.

Reviewer: ```suggestion
Your app comes with a sample [workflow](https://github.com/slack-samples/bolt-js-starter-template/blob/future/manifest/workflows/sample-workflow.js) that uses a [function](https://github.com/slack-samples/bolt-js-starter-template/blob/future/manifest/functions/sample-function.js). 

Kick off this workflow using the trigger defined in [this trigger configuration file](https://github.com/slack-samples/bolt-js-starter-template/blob/future/triggers/hello-world-trigger.json). We can run a command via the CLI to initialize this trigger so that we can run the workflow.
```

Some wording nits and maybe we break this into sections?

First, make sure you're in the project directory in your command line: `cd my-app`

Then, run the following command to create a trigger:
```
slack trigger create --trigger-def "triggers/hello-world-trigger.json"      
```

> The above command will create a link trigger, which is run by clicking on a given URL, for the selected workspace. Make sure to select the workspace you want and that it is appended by `(dev)`. This will create a dev instance of your app. Once the trigger is successfully created, you should see an output like this:

Reviewer: ```suggestion
The above command will create a link trigger belonging to the selected workspace. Make sure to select the workspace you want and that it is appended by `(dev)`. This will create a dev instance of your app. Once the trigger is successfully created, you should see an output like this:
```

I think we can get rid of the "which is run by clicking". Let's save that for the section right after.

   Trigger Name: Sample Trigger
   URL: https://slack.com/shortcuts/[ID]/[Some ID]
```

> The provided URL will be what you use to run your trigger. Copy this URL and save it somewhere; you'll need it for later.

Reviewer: ```suggestion
Copy this URL and paste it into your selected workspace. We'll be using the trigger to kick off our workflow later. 
```
